lib/singletonable.js

```diff
@@ -0,0 +1,25 @@
+// Add lazy singletons to Constructors. This adds a makeSingleton class
+// method, which will create a singleton property of the given name, passing
+// the remaining arguments to the constructor.
+//
+// The singleton is constructed on first access, through
+// Object.defineProperty. This defers initialization, simplifying dependency
+// management.
+
+module.exports = function (Constructor) {
+  Constructor.makeSingleton = function (name, varargs) {
+    // Singleton should be of whatever makeSingleton was called on, not
+    // necessarily what was passed to singletonable()
+    var Child = this;
+    varargs = arguments._.tail();
+
+    // Storage for the the singleton object, once it is created
+    var singleton;
```

Comment:
Hmmm... I guess that won't really work because of the name you're using, but you could provide a setter method